File: python/06_034.py
def compare_positions(pos1, pos2, goal):
    f1 = f_score[pos1]
    f2 = f_score[pos2]
    
    if f1 == f2:
        # Prefer positions closer to goal
        h1 = heuristic(pos1, goal)
        h2 = heuristic(pos2, goal)
        return h1 < h2
    
    return f1 < f2

File: cpp/06_034.cpp
// Helper function to get string key
std::string posToString(const std::vector<int>& pos) {
    return std::to_string(pos[0]) + "," + std::to_string(pos[1]);
}

// Dijkstra - only uses g_score
int priority = gScore[posToString(current)];

// A* - uses f_score (g_score + heuristic)
int priority = gScore[posToString(current)] + 
               heuristic(current, goal);